### PR TITLE
Adding workflow for sobelow (static analysis tool for the Phoenix framework)

### PR DESCRIPTION
**Pull request by abirismyname**
## Pre-requisites

- [ ] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [x] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [x] Should include comments in the workflow for any parts that are not obvious or could use clarification.

**For _CI_ workflows, the workflow:**

- [ ] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [ ] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [ ] Packaging workflows should run on `release` with `types: [ created ]`.
- [ ] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [x] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [x] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [x] `name`: Name of the Code Scanning integration.
  - [ ] `organization`: Name of the organization producing the Code Scanning integration.
  - [x] `description`: Short description of the Code Scanning integration.
  - [x] `categories`: Array of languages supported by the Code Scanning integration.
  - [x] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [x] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [x] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [x] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [x] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [x] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

File: code-scanning/sobelow.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  mobile-security:
+    permissions:
+      contents: read # for actions/checkout to fetch code
+      security-events: write # for github/codeql-action/upload-sarif to upload SARIF results
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Elixir
+        uses: erlef/setup-beam@988e02bfe678367a02564f65ca2e37726dc0268f
```
Comment (Phantsure) [inline code-scanning/sobelow.yml:29]: What is the reason behind not using the latest release?

File: code-scanning/properties/sobelow.properties.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+    "name": "Sobelow",
+    "creator": "nccgroup",
+    "description": "Sobelow is a security-focused static analysis tool for the Phoenix framework.",
+    "iconName": "sobelow",
+    "categories": [
+      "Code Scanning",
+      "Elixir"
```
Comment (bishal-pdMSFT) [inline code-scanning/properties/sobelow.properties.json:8]: Do you want to add `Phoenix` here as well? If there are files specific to this framework, we can enhance our starter workflows recommendation logic by detecting such files.

Comment (bishal-pdMSFT) [reply code-scanning/properties/sobelow.properties.json:8]: If yes, please create an issue here using "Support new tech_stack" issue template.

File: code-scanning/sobelow.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
```
Comment (bishal-pdMSFT) [inline code-scanning/sobelow.yml:4]: Please add some comments about what this workflow does and if there are any pre-requisites.
Also, add links for user to read more or get support.

File: code-scanning/properties/sobelow.properties.json (hunk 2 above)
Comment (abirismyname) [reply code-scanning/properties/sobelow.properties.json:8]: `Phoenix` wasn't listed in [linguist](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml) so I didn't add it as a category.

File: code-scanning/sobelow.yml (hunk 1 above)
Comment (abirismyname) [reply code-scanning/sobelow.yml:29]: I got the sha from the elixir starter [workflow](https://github.com/actions/starter-workflows/blob/main/ci/elixir.yml). I'll update it to 1.11.

Comment (abirismyname) [reply code-scanning/sobelow.yml]: Turns out this step isn't necessary so I removed it.

**Comment by abirismyname** [comment]
@bishal-pdMSFT @Phantsure Changes made. I also made a test repo where you can see the scan results here:
https://github.com/shundor/phoenix-hello-world/security/code-scanning

File: code-scanning/sobelow.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+#
+# Sobelow is a security-focused static analysis tool for the Phoenix framework. https://sobelow.io/
+#
+# To use this workflow, you must have GitHub Advanced Security (GHAS) enabled for your repository.
+#
+# Instructions:
+# 1. Update elixir-version and otp-version in the "Set up Elixir" action below.
```
Comment (Phantsure) [inline code-scanning/sobelow.yml:11]: This can be updated since the actions is removed.

Comment (abirismyname) [reply code-scanning/sobelow.yml]: Fixed

**Review by bishal-pdMSFT — APPROVED** [review]

**Pull request merged**